cmake/HalideTestHelpers.cmake

```diff
@@ -49,9 +49,15 @@ function(add_halide_test TARGET)
              COMMAND ${args_COMMAND} ${args_ARGS}
              WORKING_DIRECTORY "${args_WORKING_DIRECTORY}")
 
+    # If the test is not marked as being multi-threaded,
+    # limit it to one thread, as requested.
+    if (NOT "multithreaded" IN_LIST args_GROUPS)
+        set(MAYBE_FORCE_NO_THREADING "HL_NUM_THREADS=1")
+    endif()
+
```

Comment:
I think we don't want this here. With the labels in place, you can always run:

```
$ HL_NUM_THREADS=1 ctest -LE multithreaded 
$ ctest -L multithreaded
```

But with this here, you _can't_ run the unmarked tests with multiple threads.

Reply:
> With the labels in place, you can always run:
> 
> ```
> $ HL_NUM_THREADS=1 ctest -LE multithreaded 
> $ ctest -L multithreaded
> ```

Hm, that's a good point, thanks!

> you can't run the unmarked tests with multiple threads.

That's actually The point, to ensure that there are no false-negatives ("uses threads but not marked as such"),
by essentially requiring to explicitly opt-in into using threads.

I can drop this of course, but i think this is an important part to ensure lack of regressions.